### PR TITLE
CI: Upload Reports to github actions

### LINKED ISSUE #2486: CI: Upload CI check Reports to GH-Actions
**Is your feature request related to a problem? Please describe.**
Upload the reports of following to GH-Actions:
* Static Analysis
* UnitTests' Report from Codecov
* Instrumentation Tests' Report from Codecov

**Describe the solution you'd like**
New jobs/steps at concerned sites.

**Describe alternatives you've considered**
No other alternatives I have currently.

**Additional context**
To enhance CI process

### PR DESCRIPTION
**Pull request by s-ayush2903**
Fixes #2486 

Added concerned jobs and steps wherever needed

**Screenshots** 
NA

### CODE REVIEW

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -45,3 +45,9 @@ jobs:
       - name: Upload coverage to Codecov
         run: |
           bash <(curl -s https://codecov.io/bash)
+
+      - name: Upload Coverage to GH-Actions
+        uses: actions/upload-artifact@v2.2.0
+        with:
+          name: Tests Coverage Report
+          path: app/build/reports/
```
Comment (macgills) [inline .github/workflows/coverage.yml:53]: wildcards needed here too to get custom coverage

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -16,6 +16,14 @@ jobs:
       - name: Static Analysis
         run: ./gradlew ktlintCheck detekt app:lintDebug custom:lintCustomexampleDebug
 
+      - name: Upload Static Analysis Report
+        uses: actions/upload-artifact@v2.2.0
+        with:
+          name: Static Analysis Report
+          path: |
+            **/**/reports/
```
Comment (macgills) [inline .github/workflows/pull_request.yml:24]: you shouldn't need 2 `**`, 1 `**` means "any depth of directory"

File: .github/workflows/coverage.yml [hunk 3]
```diff
@@ -45,3 +45,10 @@ jobs:
       - name: Upload coverage to Codecov
         run: |
           bash <(curl -s https://codecov.io/bash)
+
+      - name: Upload Coverage to GH-Actions
+        uses: actions/upload-artifact@v2.2.0
+        with:
+          name: Tests Coverage Report
+          path: |
+            **/*/reports/
```
Comment (macgills) [inline .github/workflows/coverage.yml:54]: we shouldn't need the `*` here?

Comment (s-ayush2903) [reply .github/workflows/coverage.yml:54]: Ah shoot!

**Review by macgills — DISMISSED** [review]

**Comment by s-ayush2903** [comment]
I'm squashing 'em, too many commits for minute changes.

**Review by macgills — APPROVED** [review]

**Pull request merged**